src/taipy/core/data/data_node.py

```diff
@@ -436,11 +437,16 @@ def filter(self, operators: Union[List, Tuple], join_operator=JoinOperator.AND):
         if not ((type(operators[0]) == list) or (type(operators[0]) == tuple)):
             if isinstance(data, (pd.DataFrame, modin_pd.DataFrame)):
                 return DataNode.__filter_dataframe_per_key_value(data, operators[0], operators[1], operators[2])
+            if isinstance(data, np.ndarray):
+                list_operators = [operators]
+                return DataNode.__filter_numpy_array(data, list_operators)
             if isinstance(data, List):
                 return DataNode.__filter_list_per_key_value(data, operators[0], operators[1], operators[2])
         else:
             if isinstance(data, (pd.DataFrame, modin_pd.DataFrame)):
                 return DataNode.__filter_dataframe(data, operators, join_operator=join_operator)
+            if isinstance(data, np.ndarray):
+                return DataNode.__filter_numpy_array(data, operators, join_operator=join_operator)
             if isinstance(data, List):
                 return DataNode.__filter_list(data, operators, join_operator=join_operator)
```

Comment:
The `datanode.filter` method could delegate the filtering logic to a dedicated class. What do you think? is n't it the purpose of the `_FilterDataNode` class?